internal/controller/aggregates_controller.go

```
@@ -127,6 +118,23 @@ func (ac *AggregatesController) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, ac.Status().Update(ctx, hv)
 }
 
+func (ac *AggregatesController) errorCondition(ctx context.Context, hv *kvmv1.Hypervisor, err error) (ctrl.Result, error) {
+	condition := metav1.Condition{
+		Type:    ConditionTypeAggregatesUpdated,
+		Status:  metav1.ConditionFalse,
+		Reason:  ConditionAggregatesFailed,
+		Message: err.Error(),
+	}
+
+	if meta.SetStatusCondition(&hv.Status.Conditions, condition) {
+		if err2 := ac.Status().Update(ctx, hv); err2 != nil {
+			return ctrl.Result{}, err2
+		}
+	}
+
+	return ctrl.Result{RequeueAfter: defaultWaitTime}, nil
+}
```

I would change the signature to return only error:

1. You can return the result of ac.Status().Update() directly
2. It's more reusable and better readable inside the reconciliation loop that you return a ReturnAfter Result after an error of `errorCondition`.
3. it's possible to encapsulate it into a Retry

---

This is kind of outdated with #160 being merged, so I need to revisit it.

> 1. You can return the result of ac.Status().Update() directly

We filter now by the generation-changed, so we do not reconcile again when we update the status.
So, if we return there no error, the reconciliation is finished and will never succeed.
I swallowed the error, because I understood you before #160, you didn't want the openstack errors to fill the logs.

With #160 merged, I can return the error, do need to return a `RequeueAfter`